src/sentry/integrations/utils/metrics.py

```diff
@@ -25,65 +24,86 @@ def __str__(self) -> str:
+        """Return the domain that the integration belongs to."""
         raise NotImplementedError
 
-    @staticmethod
-    def get_standard_key(
-        domain: str,
-        integration_name: str,
-        interaction_type: str,
-        outcome: EventLifecycleOutcome,
-        *extra_tokens: str,
-    ) -> str:
-        """Construct a key with a standard cross-integration structure.
-
-        Implementations of `get_key` generally should delegate to this method in
-        order to ensure consistency across integrations.
-
-        :param domain:           a constant string representing the category of business
-                                 concern or vertical domain that the integration belongs
-                                 to (e.g., "messaging" or "source_code_management")
-        :param integration_name: the name of the integration (generally should match a
-                                 package name from `sentry.integrations`)
-        :param interaction_type: a key representing the category of interaction being
-                                 captured (generally should come from an Enum class)
-        :param outcome:          the object representing the event outcome
-        :param extra_tokens:     additional tokens to add extra context, if needed
-        :return: a key to represent the event in metrics or logging
+    @abstractmethod
+    def get_integration_name(self) -> str:
+        """Return the name of the integration.
+
+        This value generally should match a package name from `sentry.integrations`.
         """
+        raise NotImplementedError
 
-        # For now, universally include an "slo" token to distinguish from any
-        # previously existing metrics keys.
-        # TODO: Merge with or replace existing keys?
-        root_tokens = ("sentry", "integrations", "slo")
+    @abstractmethod
+    def get_interaction_type(self) -> str:
+        """Return a key representing the category of interaction being captured.
 
-        specific_tokens = (domain, integration_name, interaction_type, str(outcome))
-        return ".".join(itertools.chain(root_tokens, specific_tokens, extra_tokens))
+        Generally, this string value should always come from an instance of an Enum
+        class. But each subclass can define its own Enum of interaction types and
+        there is no strict contract that relies on the Enum class.
+        """
 
-    def get_extras(self) -> Mapping[str, Any]:
-        """Get extra data to log."""
-        return {}
+        raise NotImplementedError
 
-    def capture(self, assume_success: bool = True) -> "EventLifecycle":
-        """Open a context to measure the event."""
-        return EventLifecycle(self, assume_success)
+    def get_metric_id(self, outcome: EventLifecycleOutcome) -> MetricId:
+        """Construct the metrics key and tags that will represent this event."""
+
+        key_tokens = (
+            "sentry",
```

Comment:
This shows up as "sentry.sentry.integrations.*" in Datadog so we probably want to remove this first "sentry" while we're making changes.

Reply:
Removed.